src/plots/array_container_defaults.js

```diff
@@ -64,4 +67,13 @@ module.exports = function handleArrayContainerDefaults(parentObjIn, parentObjOut
 
         contOut.push(itemOut);
     }
+
+    // in case this array gets its defaults rebuilt independent of the whole layout,
+    // relink the private keys just for this array.
+    if(Lib.isArray(previousContOut)) {
+        var len = Math.min(previousContOut.length, contOut.length);
+        for(i = 0; i < len; i++) {
+            Lib.relinkPrivateKeys(contOut[i], previousContOut[i]);
+        }
+    }
```

Comment:
https://github.com/plotly/plotly.js/pull/1403/commits/70c882bceaa5236b7c1e7cdf51903c5d1823449b keeps the private keys in sync to the degree that it can - which is unless you delete elements from the array. Then they get lost. @rreusser as this isn't new behavior are you OK with it as it stands now?